llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -1275,7 +1294,7 @@ void InferAddressSpacesImpl::performPointerReplacement(
     // This instruction may contain multiple uses of V, update them all.
     CurUser->replaceUsesOfWith(
         V, new AddrSpaceCastInst(NewV, V->getType(), "", InsertPos));
-  } else {
+  } else if (isa<Constant>(V)) {
     CurUserI->replaceUsesOfWith(
         V, ConstantExpr::getAddrSpaceCast(cast<Constant>(NewV), V->getType()));
   }
```

Comment:
we should probably add `else llvm_unreachable()` if all expected cases are handled above.

Otherwise it's not clear why there was a catch-all handling before, but not now.

Reply:
Good point, I've switched this back to catch-all handling and updated the above `if` branch to handle arguments. I'm surprising the previous version seemed to work at all 😅